src/library/scala/MatchError.scala

```diff
@@ -19,11 +21,11 @@ package scala
  *  @version 1.1, 05/03/2004
  *  @since   2.0
  */
-final class MatchError(obj: Any) extends RuntimeException {
+final class MatchError(@transient obj: Any) extends RuntimeException {
   /** There's no reason we need to call toString eagerly,
    *  so defer it until getMessage is called.
    */
-  private lazy val objString = {
+  private val objString = {
```

Comment:
Is there any reason why a `lazy val` wouldn't still work? I did this (on 2.12.1) and the test still passed, and the bytecode appears unobjectionable. It makes this a very small change.

```
final class ScratchError(@transient obj: Any) extends RuntimeException with Serializable {
  private lazy val message = {
    def ofClass = "of class " + obj.getClass.getName
    if (obj == null) "null"
    else try {
      obj.toString + " (" + ofClass + ")"
    } catch {
      case _: Throwable => "an instance " + ofClass
    }
  }
  @throws[java.io.ObjectStreamException]
  private def writeReplace(): Object = {
    message // force eval
    this
  }
  override def getMessage = message
}
```